src/gwt/src/org/rstudio/studio/client/workbench/views/files/FilesUpload.java

```diff
@@ -67,22 +67,27 @@ public void execute()
          new OperationWithInput<PendingFileUpload>() {
             public void execute(PendingFileUpload pendingUpload)
             {
-               // confirm overwrites if necessary
+               Boolean unzipFound = pendingUpload.getUnzipFound();
+               Boolean isZip = pendingUpload.getIsZip();
                FileUploadToken token = pendingUpload.getToken();
-               if (pendingUpload.getOverwrites().length() > 0)
+
+               // confirm unzip is installed
+               if (unzipFound == Boolean.FALSE && isZip == Boolean.TRUE)
                {
+                  // Warn user unzip is not installed
                   globalDisplay_.showYesNoMessage(
-                       MessageDialog.WARNING, 
-                       "Confirm Overwrite",
-                       confirmFileUploadOverwriteMessage(pendingUpload), 
-                       false, 
-                       completeFileUploadOperation(token, true), 
-                       completeFileUploadOperation(token, false), 
-                       false);
+                          MessageDialog.WARNING,
+                          "Unzip not found on PATH",
+                          "Unzip was not found on users PATH. Would you like to upload the zip archive without unzipping?",
```

Comment:
For error / warning messages, I think a good format to adopt is:

1. What happened,
2. Why this matters,
3. What do you want to do

```suggestion
                          "The unzip system utility could not be found. unzip is required for decompressing .zip archives after upload. Would you like to upload the zip archive without unzipping?",
```